core/src/main/java/com/sequenceiq/cloudbreak/service/cluster/flow/AmbariHostsJoinStatusCheckerTask.java

```diff
@@ -14,14 +19,25 @@
 
     private static final Logger LOGGER = LoggerFactory.getLogger(AmbariHostsJoinStatusCheckerTask.class);
 
+    @Inject
+    private HostMetadataRepository hostMetadataRepository;
+
     @Override
     public boolean checkStatus(AmbariHostsCheckerContext hosts) {
+        Set<HostMetadata> allInStack = hostMetadataRepository.findHostsInCluster(hosts.getStack().getCluster().getId());
         try {
             AmbariClient ambariClient = hosts.getAmbariClient();
             List<String> hostNames = ambariClient.getClusterHosts();
-            for (String hostName : hostNames) {
-                if ("UNKNOWN".equals(ambariClient.getHostState(hostName))) {
-                    LOGGER.info("The state of the {} is UNKNOWN, waiting for join", hostName);
+            for (HostMetadata hostMetadata : allInStack) {
+                boolean contains = false;
```

Comment:
This logic is invalid. getClusterHosts will always return the hosts in the cluster no matter they're joined or not. You need to use getHostStatuses which will return all the hosts and their states then check if the hosts from metadata have any state other than UNKNOWN.